File: src/main/java/me/miki/shindo/management/settings/impl/TextSetting.java
package me.miki.shindo.management.settings.impl;

import me.miki.shindo.management.settings.config.ConfigOwner;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.settings.Setting;

public class TextSetting extends Setting {

    private final String defaultText;
    private String text;

    public TextSetting(TranslateText tText, ConfigOwner parent, String text) {
        super(tText, parent);
        this.text = text;
        this.defaultText = text;
    }

    public TextSetting(String name, ConfigOwner parent, String text) {
        super(name, parent);
        this.text = text;
        this.defaultText = text;
    }

    @Override
    public void reset() {
        this.text = defaultText;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getDefaultText() {
        return defaultText;
    }
}

File: src/main/kotlin/me/miki/shindo/management/settings/impl/TextSetting.kt
package me.miki.shindo.management.settings.impl

import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.management.settings.Setting
import me.miki.shindo.management.settings.config.ConfigOwner

open class TextSetting : Setting {
    private val defaultText: String
    private var text: String

    constructor(text: TranslateText, parent: ConfigOwner, value: String) : super(text, parent) {
        this.text = value
        this.defaultText = value
    }

    constructor(name: String, parent: ConfigOwner, value: String) : super(name, parent) {
        this.text = value
        this.defaultText = value
    }

    override fun reset() {
        text = defaultText
    }

    fun getText(): String = text

    open fun setText(text: String) {
        this.text = text
    }

    fun getDefaultText(): String = defaultText
}
